homeassistant/components/sensor/forecast.py

```diff
@@ -124,80 +116,66 @@ def state(self):
-            elif self.type == 'pressure':
-                self._state = round(data.pressure, 1)
-            elif self.type == 'visibility':
-                self._state = data.visibility
-            elif self.type == 'ozone':
-                self._state = round(data.ozone, 1)
-
-        except forecastio.utils.PropertyUnavailable:
-            pass
+        # Call the API for new forecast data. Each sensor will re-trigger this
+        # same exact call, but thats fine. We cache results for a short period
+        # of time to prevent hitting API limits. Note that forecast.io will
+        # charge users for too many calls in 1 day, so take care when updating.
+        self.forecast_data.update()
+
+        if self.type == 'minutely_summary':
+            self.forecast_data.update_minutely()
+            minutely = self.forecast_data.data_minutely
+            self._state = getattr(minutely, 'summary', '')
+        elif self.type == 'hourly_summary':
+            self.forecast_data.update_hourly()
+            hourly = self.forecast_data.data_hourly
+            self._state = getattr(hourly, 'summary', '')
+        elif self.type == 'daily_summary':
+            self.forecast_data.update_daily()
+            daily = self.forecast_data.data_daily
+            self._state = getattr(daily, 'summary', '')
+        else:
+            self.forecast_data.update_currently()
+            currently = self.forecast_data.data_currently
+            self._state = self.get_currently_state(currently)
+
+    def get_currently_state(self, data):
+        """
+        Helper function that returns a new state based on the type.
+
+        If the sensor type is unknown, the current state is returned.
+        """
+        return {
+            'summary': getattr(data, 'summary', ''),
+            'icon': getattr(data, 'icon', ''),
+            'nearest_storm_distance': getattr(data, 'nearestStormDistance', 0),
+            'nearest_storm_bearing': getattr(data, 'nearestStormBearing', 0),
+            'precip_intensity': getattr(data, 'precipIntensity', 0),
+            'precip_type': getattr(data, 'precipType', ''),
+            'precip_probability': round(
+                getattr(data, 'precipProbability', 0) * 100, 1),
+            'dew_point': round(getattr(data, 'dewPoint', 0), 1),
+            'temperature': round(getattr(data, 'temperature', 0), 1),
+            'apparent_temperature': round(
+                getattr(data, 'apparentTemperature', 0), 1),
+            'wind_speed': getattr(data, 'windSpeed', 0),
+            'wind_bearing': getattr(data, 'windBearing', 0),
+            'cloud_cover': round(getattr(data, 'cloudCover', 0) * 100, 1),
+            'humidity': round(getattr(data, 'humidity', 0) * 100, 1),
+            'pressure': round(getattr(data, 'pressure', 0), 1),
+            'visibility': getattr(data, 'visibility', 0),
+            'ozone': round(getattr(data, 'ozone', 0), 1)
+        }.get(self.type, self._state)
```

Comment:
This now works. catching `PropertyUnavailable ` no longer relevant.

Two things to discuss:
1) I thought about converting snake case to camel case to further simplify the lookup. Problem is that half of the responses are treated like integers and need different operations done on them.
2) defaults. For the numbers, defaulting to zero works fine, but I imagine there could be an edge case where an api error results in a return that there is "zero percent chance of rain". Suddenly someone home automation setup kicks off the sprinklers during a downpour. Not sure if there are any thoughts on that.